pkgs/build-support/setup-hooks/win-dll-link.sh

```diff
@@ -1,15 +1,47 @@
 
 fixupOutputHooks+=(_linkDLLs)
 
-# For every *.{exe,dll} in $output/bin/ we try to find all (potential)
-# transitive dependencies and symlink those DLLs into $output/bin
-# so they are found on invocation.
+addEnvHooks "$targetOffset" linkDLLGetFolders
+
+linkDLLGetFolders() {
+    addToSearchPath "LINK_DLL_FOLDERS" "$1/lib"
+    addToSearchPath "LINK_DLL_FOLDERS" "$1/bin"
+}
+
+_linkDLLs() {
+    linkDLLsInfolder "$prefix/bin"
+}
+
+# Try to links every known dependancy of exe/dll in the folder of the 1str input
+# into said folder, so they are found on invocation.
 # (DLLs are first searched in the directory of the running exe file.)
 # The links are relative, so relocating whole /nix/store won't break them.
-_linkDLLs() {
+linkDLLsInfolder() {
 (
-    if [ ! -d "$prefix/bin" ]; then exit; fi
-    cd "$prefix/bin"
+    local folder
+    folder="$1"
+    if [ ! -d "$folder" ]
+    then
+        echo "Not linking DLLs in the non-existant folder $folder"
+        return
+    fi
+    cd "$folder"
+
+    # Use associative arrays as set
+    local filesToChecks
+    local filesDone
+    declare -A filesToChecks # files that still needs to have their dependancies checked
+    declare -A filesDone # files that had their dependancies checked and who is copied to the bin folder if found
+
+    markFileAsDone() {
+        if [ "!${filesDone[$1]+a}" ]; then filesDone[$1]=a ; fi
```

Comment:
@marius851000 I guess you meant 
```suggestion
        if [ ! "${filesDone[$1]+a}" ]; then filesDone[$1]=a ; fi
```
here and below?

Reply:
@brano543 said the shellcheck commit broke something, maybe this caused it.

Reply:
you can apply this to your PR https://github.com/NixOS/nixpkgs/commit/25d7d273ddc2f3ce218bb8f24aaa763aedbc62b4.patch